src/main/java/org/terasology/dialogs/action/ChangeDialogActionTypeHandler.java

```diff
@@ -16,34 +16,22 @@
 
 package org.terasology.dialogs.action;
 
-import java.util.Map;
+import java.util.Optional;
 
-import org.terasology.persistence.typeHandling.DeserializationContext;
-import org.terasology.persistence.typeHandling.PersistedData;
-import org.terasology.persistence.typeHandling.PersistedDataMap;
-import org.terasology.persistence.typeHandling.RegisterTypeHandler;
-import org.terasology.persistence.typeHandling.SerializationContext;
-import org.terasology.persistence.typeHandling.SimpleTypeHandler;
-
-import com.google.common.collect.ImmutableMap;
+import org.terasology.persistence.typeHandling.*;
 
 @RegisterTypeHandler
-public class ChangeDialogActionTypeHandler extends SimpleTypeHandler<ChangeDialogAction> {
+public class ChangeDialogActionTypeHandler extends TypeHandler<ChangeDialogAction> {
 
     @Override
-    public PersistedData serialize(ChangeDialogAction action, SerializationContext context) {
-        Map<String, PersistedData> data = ImmutableMap.of(
-                "type", context.create(action.getClass().getSimpleName()),
-                "prefab", context.create(action.getPrefab()));
-
-        return context.create(data);
+    protected PersistedData serializeNonNull(ChangeDialogAction value, PersistedDataSerializer serializer) {
+        return serializer.serialize(value.getPrefab());
     }
 
     @Override
-    public ChangeDialogAction deserialize(PersistedData data, DeserializationContext context) {
+    public Optional<ChangeDialogAction> deserialize(PersistedData data) {
         PersistedDataMap root = data.getAsValueMap();
         String prefab = root.get("prefab").getAsString();
```

Comment:
Since you are only serializing the `prefab` string, `data` will not be a value map any more -- it will simply be a string. `String prefab = data.getAsString()` should work. Also, make sure to add a failing condition if the `PersistedData` is not as expected. For this case, you might want to do something like
```java
if (!data.isString()) {
    // Optional.empty() signifies that the data could not be deserialized
    return Optional.empty()
}
```

Without this condition, `getAsString` will throw an exception and the program may crash.